src/xml_document.cc

```diff
@@ -335,144 +350,112 @@ xmlParserOption getParserOptions(v8::Local<v8::Object> props) {
 
     ret |= getParserOption(props, "big_lines", XML_PARSE_BIG_LINES);        // 4194304: Store big lines numbers in text PSVI field
 
-    return (xmlParserOption)ret;
+    return ret;
 }
 
-NAN_METHOD(XmlDocument::FromHtml)
-{
-    Nan::HandleScope scope;
+xmlDoc* parseString(char* string, int length, char* baseUrl,
+                    char* encoding, int parserOptions, bool isHTML) {
+        if (isHTML) {
+            return htmlReadMemory(string, length, baseUrl, encoding, parserOptions);
+        }
+        else {
+            return xmlReadMemory(string, length, baseUrl, encoding, parserOptions);
+        }
+}
+
+xmlDoc* parseData(v8::Local<v8::Value> data, v8::Local<v8::Object> options, bool isHTML) {
+    char* baseUrl = getStringOption(
```

Comment:
pretty sure this is a memory leak. You allocate inside of getStringOption but baseUrl is not freed.